cloud-core/src/main/java/cloud/commandframework/arguments/parser/ArgumentParser.java

```diff
@@ -139,6 +141,23 @@ default int getRequestedArgumentCount() {
         return DEFAULT_ARGUMENT_COUNT;
     }
 
+    /**
+     * Returns the suggestion provider.
+     * <p>
+     * By default, this will return the parser, if the parser is also a {@link SuggestionProvider}.
+     * Otherwise, {@link SuggestionProvider#noSuggestions()} will be returned.
+     *
+     * @return the suggestion provider
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    default @NonNull SuggestionProvider<C> suggestionProvider() {
+        if (this instanceof SuggestionProvider) {
+            return (SuggestionProvider<C>) this;
+        }
+        return SuggestionProvider.noSuggestions();
```

Comment:
Would it be better if all all implementations of ArgumentParser could just implement it as returning this? I think that's better than the default implementation doing this. Parsers should always have suggestions and having a default impl doesn't force an implementation.

Reply:
Parsers have always defaulted to no suggestions, they are parsers, not suggesters. They can provide fallback suggestions when no suggestion provider is explicitly set for a component. And no, I don't think it would be better to require implementing an extra method on every parser that wants to provide those fallback suggestions.